src/main/java/im/status/keycard/KeycardApplet.java

```diff
@@ -1225,13 +1235,24 @@ private void generateKey(APDU apdu) {
     // Save seed exportability (this also indicates the seed as valid)
     masterSeedStatus = exportability; // Only save seed exportability after seed successfully loaded
 
+    // Build response [KeyUID, (optional) Seed]
+    short off = SecureChannel.SC_OUT_OFFSET;
+    apduBuffer[off++] = TLV_KEY_UID;
+    apduBuffer[off++] = KEY_UID_LENGTH;
+    generateKeyUID(apduBuffer, off);
+    off += KEY_UID_LENGTH;
+
     // Only return the seed if export is allowed on creation
     if( apduBuffer[ISO7816.OFFSET_P1] == GENERATE_KEY_P1_EXPORTABLE_ONCE ||
         apduBuffer[ISO7816.OFFSET_P1] == GENERATE_KEY_P1_EXPORTABLE_ALWAYS ) {
-      secureChannel.respond(apdu, BIP39_SEED_SIZE, ISO7816.SW_NO_ERROR);
-    } else {
-      secureChannel.respond(apdu, (short) 0, ISO7816.SW_NO_ERROR);
-    }
+        apduBuffer[off++] = TLV_SEED;
+        apduBuffer[off++] = BIP39_SEED_SIZE;
+        JCSystem.beginTransaction();
+        Util.arrayCopy(masterSeed, (short) 0, apduBuffer, off, BIP39_SEED_SIZE);
+        JCSystem.commitTransaction();
+        off += BIP39_SEED_SIZE;
+    }
+    secureChannel.respond(apdu, (short) (off - SecureChannel.SC_OUT_OFFSET), ISO7816.SW_NO_ERROR);
```

Comment:
What's with the `off - SecureChannel.SC_OUT_OFFSET`?

Reply:
The payload starts at `SecureChannel.SC_OUT_OFFSET`, so the payload length is `off - SecureChannel.SC_OUT_OFFSET`

Reply:
Sorry, was thinking it was an index 🤓